File: frontend/CommunicationLib/Model/Exception/LogicExceptions/PersistenceExceptions/UserNotExistentException.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CommunicationLib.Exception
{
    /// <summary>
    /// Exceptions for errors concerning user operations, if the user doesn't exist. This is a child of NotExistentException.
    /// </summary>
    public class UserNotExistentException : NotExistentException
    {
        /// <summary>
        /// Each Exception has an number, here it is 11251.
        /// </summary>
        new public int number { get { return _number; } }
        private int _number = 11251;
        
        /// <summary>
        /// Default constructor.
        /// </summary>
        public UserNotExistentException()
            : base("Der Nutzer existiert nicht.")
        {
        }

        /// <summary>
        /// This constructor allows to add a spezial message.
        /// </summary>
        /// <param name="message">the message</param>
        public UserNotExistentException(string message)
         : base(message)
        {
        }

        /// <summary>
        /// This constructor allows to add a spezial message and an other exception.
        /// </summary>
        /// <param name="message">the message</param>
        /// <param name="inner">the other exception</param>
        public UserNotExistentException(string message, System.Exception inner)
            : base(message, inner)
        {
        }
    }
}

File: documentation/server/backend/javaserver/src/main/java/de/hsrm/swt02/persistence/exceptions/UserNotExistentException.java
package de.hsrm.swt02.persistence.exceptions;

/**
 * Exception dealing with whenever a user is requested and it's not there.
 */
public class UserNotExistentException extends NotExistentException {

    private static final long serialVersionUID = -7867137682701372781L;
    public static final int ERRORCODE = 11251;
    
    /**
     * Constructor for the Exception.
     */
    public UserNotExistentException() {
        super("Der Nutzer existiert nicht.");
    }

    /**
     * Method dealing with the error message.
     * @param msg is the error message
     */
    public UserNotExistentException(String msg) {
        super(msg);
    }
    
    /**
     * ERRORCODE Getter.
     * @return ERRORCODE is the code
     */
    public int getErrorCode() {
        return UserNotExistentException.ERRORCODE;
    }
}
